frontend/src/App.js

```diff
@@ -1,31 +1,30 @@
 import React from 'react';
-import Product from './components/Product';
-import data from './data';
+import { BrowserRouter, Route } from 'react-router-dom';
+import HomeScreen from './screens/HomeScreen';
+import ProductScreen from './screens/ProductScreen';
+
 function App() {
   return (
-    <div className="grid-container">
-      <header className="row">
-        <div>
-          <a className="brand" href="/">
-            amazona
-          </a>
-        </div>
-        <div>
-          <a href="/cart">Cart</a>
-          <a href="/signin">Sign In</a>
-        </div>
-      </header>
-      <main>
-        <div>
-          <div className="row center">
-            {data.products.map((product) => (
-              <Product key={product._id} product={product}></Product>
-            ))}
+    <BrowserRouter>
+      <div className="grid-container">
+        <header className="row">
+          <div>
+            <a className="brand" href="/">
+              amazona
+            </a>
+          </div>
+          <div>
+            <a href="/cart">Cart</a>
+            <a href="/signin">Sign In</a>
           </div>
-        </div>
-      </main>
-      <footer className="row center">All right reserved</footer>
-    </div>
+        </header>
+        <main>
+          <Route path="/product/:id" component={ProductScreen}></Route>
```

Comment:
How would you write this line of code in v6 of react?

Reply:
I have same problem . when start npm start a problem is {Error: A <Route> is only ever to be used as the child of <Routes> element, never rendered directly. Please wrap your <Route> in a <Routes>.}

Reply:
TRY THIS...
In declaration add

import { BrowserRouter, Routes, Route } from 'react-router-dom' ;

Then add an opening and closing Router Tag like this after main opening tag

</header>
       <main>
       <Routes>
         <Route path="/product/:id" component={ProductScreen}></Route>
         <Route path="/" component={HomeScreen} exact></Route>
       </Routes>
       </main>

Reply:
USE THIS...

define in header

import { BrowserRouter, Routes, Route } from 'react-router-dom';

</header>
  <main>
    <Routes>
         <Route path="/product/:id" component={ProductScreen}></Route>
         <Route path="/" component={HomeScreen} exact></Route>
       </Routes>
       </main>

Reply:
</header>
       <main>
       <Routes>
         <Route path="/product/:id" component={ProductScreen}></Route>
         <Route path="/" component={HomeScreen} exact></Route>
       </Routes>
       </main>

Reply:
then use an opening <Routes> tag after the opening <main> and a closing </Routes> tag before the closing </main>

Reply:
@</header>
       <main>
       <Routes>
         <Route path="/product/:id" component={ProductScreen}></Route>
         <Route path="/" component={HomeScreen} exact></Route>
       </Routes>
       </main>

Reply:
Then Use an opening "Routes" tag after opening main tag and a closing  "Routes" and before the closing main tag

Reply:
![Screen Shot 2021-11-18 at 7 32 36 PM](https://user-images.githubusercontent.com/90581670/142465076-ced2e6d0-b940-4647-8a2f-d786b82cc5f6.png)

Reply:
import "routes" from react-router-dom  and  enclose route tag with routes tag

Reply:
am getting the same error..kindly forward a demo code for error correction

Reply:
It works but void of products

Reply:
`
import {BrowserRouter, Routes, Route} from "react-router-dom";

<main>
                  <Routes>
                      <Route path="/product/:id" element={<ProductScreen/>}></Route>
                      <Route path="/" element={<HomeScreen/>} exact></Route>
                  </Routes>
</main>`

hope this helps.

Reply:
> It works but void of products

```
import {BrowserRouter, Routes, Route} from "react-router-dom";

<main>
      <Routes>
          <Route path="/product/:id" element={<ProductScreen/>}></Route>
          <Route path="/" element={<HomeScreen/>} exact></Route>
      </Routes>
</main>
```

hope this works

Reply:
" Matched leaf route at location "/" does not have an element. This means it will render an <Outlet /> with a null value by default resulting in an "empty" page. "

I am getting this message after running some please tell me what is the problem and please tell what should I do to resolve it..............................

Reply:
thanks alot

Reply:
How can be get products on Homescreen

Reply:
Products does not appear on homescreen, any suggests?

Reply:
how to get product in screen

Reply:
![image](https://user-images.githubusercontent.com/113082478/203805401-854510bd-0e11-45ec-b5fa-f267e2553942.png)
For those who are using React 18 & ReactRouter v6, this is the answer that took me a day to figure out

Reply:
![image](https://user-images.githubusercontent.com/113082478/203806458-a4e88864-58db-4515-ab88-c730efd4658c.png)
For those who are using React 18 & ReactRouter v6, this is the answer that took me a day to figure out.